### DISCUSSION TITLE
how to calculate the loss?

### DISCUSSION THREAD

**Discussion by Chrisable** [pyg-team/pytorch_geometric#7612] [General]
Usually when we calculate the loss, we use the training node, such as the code below：

```
import torch.nn.functional as F

data = dataset[0]
optimizer = torch.optim.Adam(model.parameters(), lr=0.01)

for epoch in range(200):
    pred = model(data.x, data.edge_index)
    loss = F.cross_entropy(pred[data.train_mask], data.y[data.train_mask])

    # Backpropagation
    optimizer.zero_grad()
    loss.backward()
    optimizer.step()
```
But [in this paper](https://[arxiv.org/pdf/2103.02885.pdf](https://arxiv.org/pdf/2103.02885.pdf)), they use the nodes of the test set and the validation set to calculate the loss：
```
if conf['model_name'] == 'PLP':
        loss = my_loss(logp[idx_no_train], cas[-1][idx_no_train])
    else:
        loss = F.kl_div(logp, cas[-1], reduction='batchmean')
    acc_train = accuracy(logp[idx_train], labels[idx_train])
```
The complete code is as follows: https://github.com/BUPT-GAMMA/CPF/blob/master/distill_dgl.py

Is it reasonable to use test sets to participate in calculating loss?

**Comment by EdisonLeeeee** [comment]
If I understand it correctly, the test/validation set, i.e., `idx_no_train` was used for knowledge distillation, which does not involve any label information. This seems reasonable to me.

**Reply by Chrisable** [reply]
Does that cause overfitting problems?

**Reply by EdisonLeeeee** [reply]
No. Overfitting typically occurs when a model fits the `training data` too closely.

**Reply by Chrisable** [reply]
Thank you very much.

**Reply by Chrisable** [reply]
I still don't understand, can you please explain more specifically？Thank you

**Reply by EdisonLeeeee** [reply]
Sorry if my answer was not clear. Can you please let me know which part you're still unclear about?

**Comment by Chrisable** [comment]
I still don't understand, can you please explain more specifically？Thank you
&nbsp;


抓住努力的尾巴
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jintang ***@***.***&gt;; 
发送时间: 2023年6月20日(星期二) 晚上11:08
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [pyg-team/pytorch_geometric] how to calculate the loss? (Discussion #7612)





 
No. Overfitting typically occurs when a model fits the training data too closely.

**Comment by Chrisable** [comment]
是因为GCN是半监督学习，所以可以利用测试集和验证集的节点的特征信息吗&nbsp;


抓住努力的尾巴
***@***.***



&nbsp;




------------------&nbsp;原始邮件&nbsp;------------------
发件人: "Jintang ***@***.***&gt;; 
发送时间: 2023年6月21日(星期三) 下午3:44
收件人: ***@***.***&gt;; 
抄送: ***@***.***&gt;; ***@***.***&gt;; 
主题: Re: [pyg-team/pytorch_geometric] how to calculate the loss? (Discussion #7612)





 
Sorry if my answer was not clear. Can you please let me know which part you're still unclear about?

**Reply by EdisonLeeeee** [reply]
Yes but not exactly. This paper uses the predicted results (logits) from the teacher model to guide the training of the student model on unlabeled nodes (i.e., `idx_no_train`). Such an approach is feasible for both supervised and semi-supervised learning.
